librz/core/cbin.c

```diff
@@ -4484,3 +4484,87 @@ RZ_API char *rz_core_bin_method_flags_str(ut64 flags, int mode) {
 out:
 	return rz_strbuf_drain(buf);
 }
+
+RZ_API RzCmdStatus rz_core_bin_plugins_print(RzBin *bin, RzCmdStateOutput *state) {
+	RzBinPlugin *bp;
+	RzBinXtrPlugin *bx;
+	RzBinLdrPlugin *ld;
+	RzListIter *iter;
+	PJ *pj = 0;
+	if (!bin) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (state->mode == RZ_OUTPUT_MODE_JSON) {
+		pj = state->d.pj;
+		rz_cmd_state_output_array_start(state);
+	}
+	switch (state->mode) {
```

Comment:
a minor thing, but I believe it would be better to have a `rz_core_bin_plugin_print`,`rz_core_binxtr_plugin_print`,`rz_core_binldr_plugin_print` function that do something like:
```c
rz_core_bin_plugins_print(.., RzCmdStateOutput *state) {
switch (state->mode) {
...
}
}

rz_core_binxtr_plugins_print(.., RzCmdStateOutput *state) {
switch (state->mode) {
...
}
}

```

and then here in this function instead of having all this code you'd just have one:
```c
rz_list_foreach (bin->plugins, iter, bp) {
  rz_core_bin_plugin_print(bin, bp, state);
}
rz_list_foreach (bin->binxtrs, iter, bx) {
  rz_core_binxtr_plugin_print(bin, bp, state);
}
rz_list_foreach (bin->binldrs, iter, ld) {
  rz_core_binldr_plugin_print(bin, bp, state);
}
```

However, as said it's a minor thing, so don't let this stop you :) Go ahead with the other more important things!

Reply:
And I think the same thing could be done for all the other `plugins_print` functions.